rasa/shared/nlu/training_data/training_data.py

```diff
@@ -679,38 +678,54 @@ def can_train_nlu_model(self) -> bool:
         ]
         return not any([len(lst) > 0 for lst in lists_to_check])
 
-    def divide_into_chunks(self, max_size: int = 2) -> List["TrainingDataChunk"]:
+    def divide_into_chunks(self, num_chunks: int) -> List["TrainingDataChunk"]:
         """Divides the training data into smaller chunks.
 
         Each chunk should be a good representation of the complete dataset. E.g. it
         should not just contain examples of one intent, but instead match the
         distribution of the complete dataset.
 
         Args:
-            max_size: The maximum size (in MB) of one chunk.
+            num_chunks: The total number of chunks into which the training data should be broken.
 
         Returns:
             A list of all training data chunks.
         """
-        pass
+        all_chunks = []
+        data_to_chunk = copy.deepcopy(self)
 
+        for chunk_index in range(num_chunks):
 
-class TrainingDataChunk:
-    """Holds a portion of the complete TrainingData."""
+            chunk_size_fraction = 1 / (num_chunks - chunk_index)
+            current_chunk, leftover_examples = data_to_chunk.split_nlu_examples(
```

Comment:
`split_nlu_examples` might result in a split where one example is in both chunks (`current_chunk, leftover_examples`). I guess that is not a problem.
Also would it be a problem if we just have one example for an intent in a chunk? I guess not as we will train over all chunks in the end.
 Just want to be sure 🙂

Reply:
> split_nlu_examples might result in a split where one example is in both chunks (current_chunk, leftover_examples)

This shouldn't happen ideally because it is used inside `train_test_split` as well. It would result in train test leakage.

> Also would it be a problem if we just have one example for an intent in a chunk? I guess not as we will train over all chunks in the end.

A chunk can def have one example for an intent but it's only problematic if we plan to call `train` method of an ML component like DIET for each chunk separately. I don't think we intend to do that, right?

Reply:
> I don't think we intend to do that, right?

yes, that is not planned. just wanted to make sure that we don't overlook something.